crates/red_knot_python_semantic/src/types.rs

```diff
@@ -5377,6 +5427,9 @@ pub struct FunctionType<'db> {
     /// A set of special decorators that were applied to this function
     decorators: FunctionDecorators,
 
+    /// The generic context of a generic function.
+    generic_context: Option<GenericContext<'db>>,
```

Comment:
One thing that this PR is not yet modeling well is nested generic contexts.  Tests are passing for a generic method inside of a generic class, but I think we might need to actually encode nesting of `GenericContext`s to be able to accurate display e.g. the still-generic method of a specialized generic class.

Reply:
Is it worth a TODO comment somewhere to capture this?

Reply:
Done

Reply:
I added a failing test case for this too, showing how we're not correctly modeling e.g. a generic `__init__` method inside of a generic class.